src/main/java/net/neoforged/neoforge/network/filters/NetworkFilters.java

```diff
@@ -35,6 +40,22 @@ public static void injectIfNecessary(Connection manager) {
         });
     }
 
+    public static void cleanIfNecessary(Connection manager) {
+        ChannelPipeline pipeline = manager.channel().pipeline();
+        if (pipeline.get("packet_handler") == null)
+            return; // Realistically this can only ever be null if the connection was prematurely closed due to an error. We return early here to reduce further log spam.
+
+        //Grab the pipeline filters to remove in a seperate list to avoid a ConcurrentModificationException
+        final List<VanillaPacketFilter> toRemove = pipeline.names()
+                .stream()
+                .map(name -> pipeline.get(name))
```

Comment:
Instead of mapping it like this by looking it up from the pipeline for each name we got should we maybe make use of `pipeline.toMap`?

Reply:
We want to remove all elements of a specific type. Using `toMap` would not help here.